packages/theme/pages/Checkout/Billing.vue

```diff
@@ -233,18 +243,27 @@ export default {
     });
 
     const handleFormSubmit = async () => {
-      await save({ billingDetails: form.value });
+      if (isAuthenticated.value && selectedAddressId.value) {
+        const selectedAddress = billing.value.addresses.find(e => e._id === selectedAddressId.value);
+        await save({ billingDetails: selectedAddress });
+      } else {
+        await save({ billingDetails: form.value });
+      }
       context.root.$router.push('/checkout/payment');
     };
 
     onSSR(async () => {
       await load();
+      await loadUserBilling();
```

Comment:
One more thing here. Can we make it in a way, that if the checkout is already past shipping/billing step (has address attached), we would either select the option that was selected (if the address matches a saved address) or we would prefill the form?